src/tasks/block.rs

```diff
@@ -139,39 +139,60 @@ impl BlockBuilder {
         }
     }
 
+    /// Fetches transactions from the cache and ingests them into the in progress block
     async fn get_transactions(&mut self, in_progress: &mut InProgressBlock) {
-        tracing::trace!("query transactions from cache");
+        trace!("query transactions from cache");
         let txns = self.tx_poller.check_tx_cache().await;
         match txns {
             Ok(txns) => {
-                tracing::trace!("got transactions response");
+                trace!("got transactions response");
                 for txn in txns.into_iter() {
                     in_progress.ingest_tx(&txn);
                 }
             }
             Err(e) => {
-                tracing::error!(error = %e, "error polling transactions");
+                error!(error = %e, "error polling transactions");
             }
         }
     }
 
-    async fn _get_bundles(&mut self, in_progress: &mut InProgressBlock) {
-        tracing::trace!("query bundles from cache");
+    /// Fetches bundles from the cache and ingests them into the in progress block
+    async fn get_bundles(
+        &mut self,
+        ru_provider: &WalletlessProvider,
+        in_progress: &mut InProgressBlock,
+    ) {
+        trace!("query bundles from cache");
         let bundles = self.bundle_poller.check_bundle_cache().await;
+        // TODO: Sort bundles received from cache
         match bundles {
             Ok(bundles) => {
-                tracing::trace!("got bundles response");
                 for bundle in bundles {
-                    in_progress.ingest_bundle(bundle);
+                    let result = self.simulate_bundle(&bundle.bundle, ru_provider).await;
+                    if result.is_ok() {
+                        in_progress.ingest_bundle(bundle.clone());
+                    }
```

Comment:
we might want to log a simulation error if it's not OK (although this would ofc never be triggered rn)